### PR TITLE
[Merged by Bors] - chore: remove superseded `small_of_fintype`

### PR DESCRIPTION
**Pull request by TwoFX**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — COMMENTED** [review]
This already exists as `small_of_fintype` (but you can replace the `Fintype` assumption by `Finite` and rename it to the correct name of `Finite.toSmall`)

**Comment by YaelDillies** [comment]
Forgetful inheritance instances should be named `A.toB`. The rule is extended to Prop-valued typeclasses where it's not quite "inheritance" anymore.

**Comment by TwoFX** [comment]
@YaelDillies this sounds strange to me, doesn't `toSmall` break the rule that Prop-valued theorems should use camel case?

**Comment by YaelDillies** [comment]
It's not a theorem, it's an instance. That's where the difference is. See eg `GCDMonoid.toIsIntegrallyClosed`.

**Comment by TwoFX** [comment]
Okay, I've renamed it to `Finite.toSmall`.

File: Mathlib/Logic/Small/Basic.lean [hunk 1]
```diff
@@ -57,7 +57,7 @@ theorem small_of_injective_of_exists {α : Type v} {β : Type w} {γ : Type v'}
     infer_instance
 
 /-!
-We don't define `small_of_fintype` or `small_of_countable` in this file,
+We don't define `Finite.toSmall` or `small_of_countable` in this file,
 to keep imports to `Logic` to a minimum.
```
Comment (YaelDillies) [inline Mathlib/Logic/Small/Basic.lean:61]: Can you delete `small_of_countable`? It's completely superseded now
```suggestion
We don't define `Finite.toSmall` in this file to keep imports to `Logic` to a minimum.
```

Comment (TwoFX) [reply Mathlib/Logic/Small/Basic.lean:61]: Sorry, but superseded by what?

Comment (YaelDillies) [reply Mathlib/Logic/Small/Basic.lean:61]: Ah sorry, I meant the other way around. Can you try deleting `Finite.toSmall` and renaming `small_of_countable` to `Countable.toSmall`?

Comment (TwoFX) [reply Mathlib/Logic/Small/Basic.lean]: Done.

File: Mathlib/Data/Countable/Small.lean [hunk 2]
```diff
@@ -17,7 +18,13 @@ That is, any countable type is equivalent to a type in any universe.
...
   let ⟨_, hf⟩ := exists_injective_nat α
   small_of_injective hf
-#align small_of_countable small_of_countable
+#align small_of_countable Countable.toSmall
+
+@[deprecated] -- 2024-03-19
+alias small_of_countable := Countable.toSmall
+
+theorem Finite.small (α : Type v) [Finite α] : Small.{w} α := inferInstance
+#align small_of_fintype Finite.small
```
Comment (YaelDillies) [inline Mathlib/Data/Countable/Small.lean:30]:
```suggestion
#align small_of_fintype Countable.toSmall
```

File: Mathlib/Data/Countable/Small.lean [hunk 3]
```diff
@@ -17,7 +18,13 @@ That is, any countable type is equivalent to a type in any universe.
...
 
-instance (priority := 100) small_of_countable (α : Type v) [Countable α] : Small.{w} α :=
+instance (priority := 100) Countable.toSmall (α : Type v) [Countable α] : Small.{w} α :=
   let ⟨_, hf⟩ := exists_injective_nat α
   small_of_injective hf
-#align small_of_countable small_of_countable
+#align small_of_countable Countable.toSmall
+
+@[deprecated] -- 2024-03-19
+alias small_of_countable := Countable.toSmall
```
Comment (YaelDillies) [inline Mathlib/Data/Countable/Small.lean:27]: This one you can nolint

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by fpvandoorn** [comment]
Renamed the PR to reflect the changes from the review

bors merge

**Pull request closed without merging**